src/main.cpp

```diff
@@ -37,8 +38,12 @@
 #define SDL_MAIN_HANDLED
 #include <SDL.h>
 
-#if BOOST_OS_LINUX || BOOST_OS_MACOS
+#if BOOST_OS_LINUX
 #define _putenv(__s) putenv((char*)(__s))
```

Comment:
This is causing the build error on macOS, it doesn't get defined in macOS. I just replaced it outside the guards with this and it compiled just fine for all platforms. 
```
#ifndef _putenv
#define _putenv(__s) putenv((char*)(__s))
#endif
```